arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -1722,58 +1740,47 @@ ErrorCode RocksDBMetaCollection::doLock(double timeout, AccessMode::Type mode) {
   if (timeout <= 0) {
     timeout = defaultLockTimeout;
   }
-  auto timeout_us = std::chrono::duration_cast<std::chrono::microseconds>(
+  auto timeout_ms = std::chrono::duration_cast<std::chrono::milliseconds>(
       std::chrono::duration<double>(timeout));
 
-  constexpr auto detachThreshold = std::chrono::microseconds(1000000);
-
   bool gotLock = false;
-  if (timeout_us > detachThreshold) {
-    if (mode == AccessMode::Type::WRITE) {
-      gotLock = _exclusiveLock.tryLockWriteFor(detachThreshold);
-    } else {
-      gotLock = _exclusiveLock.tryLockReadFor(detachThreshold);
-    }
+  if (mode == AccessMode::Type::WRITE) {
+#if USE_FUTURE_MUTEX
+    auto result =
+        co_await asTry(_exclusiveLock.asyncTryLockExclusiveFor(timeout_ms));
+    gotLock = result.hasValue();
     if (gotLock) {
-      // keep the lock and exit
-      return TRI_ERROR_NO_ERROR;
+      result.get().release();
     }
-    LOG_TOPIC("dd231", INFO, Logger::THREADS)
-        << "Did not get lock within 1 seconds, detaching scheduler thread.";
-    uint64_t currentNumberDetached = 0;
-    uint64_t maximumNumberDetached = 0;
-    Result r = arangodb::SchedulerFeature::SCHEDULER->detachThread(
-        &currentNumberDetached, &maximumNumberDetached);
-    if (r.is(TRI_ERROR_TOO_MANY_DETACHED_THREADS)) {
-      LOG_TOPIC("dd232", WARN, Logger::THREADS)
-          << "Could not detach scheduler thread (currently detached threads: "
-          << currentNumberDetached
-          << ", maximal number of detached threads: " << maximumNumberDetached
-          << "), will continue to acquire "
-             "lock in scheduler thread, this can potentially lead to "
-             "blockages!";
-    }
-  }
-
-  timeout_us -= detachThreshold;
-
-  if (mode == AccessMode::Type::WRITE) {
-    gotLock = _exclusiveLock.tryLockWriteFor(timeout_us);
+#else
+    gotLock = _exclusiveLockOld.tryLockWriteFor(
```

Comment:
If we need this case, we should go back to the thread detaching. If not, the case should be removed.